### ISSUE TITLE
document.update() sends deprecated 'update' command.

### ISSUE THREAD

**Issue by bdentino** [Automattic/mongoose#8182]
**Do you want to request a *feature* or report a *bug*?**
Request a feature (more like a minor enhancement)

**What is the current behavior?**
The document `doc.update()` method issues a node deprecation warning: 
```
(node:100) DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

> Not a bug

**What is the expected behavior?**

I understand what the warning is for, and it makes sense to me that this should be issued when doing a similar operation using the static `Model.update(...)`, since this method can operate on more than one document. However, when we are operating on a single mongoose document instance `doc`, `doc.update()` necessarily implies `collection.updateOne`. Since it appears that the Document instance method `update()` will continue to be supported, it would be nice if the implementation were modified to issue an `updateOne` command under the hood automatically to avoid the deprecation warning.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

Mongoose v5.7.1
MongoDB v4.2
Node.js v11.10.1

**Comment by vkarpov15** [comment]
Why not just switch to using `updateOne()` or `updateMany()`? We can't really switch `update()` to use `updateOne()`, because `update()` may update one or many documents depending on the [`multi` option](https://mongoosejs.com/docs/api/model.html#model_Model.update).

We are planning on removing `update()` in a future release, so please switch to using [`Document#updateOne()`](https://mongoosejs.com/docs/api/document.html#document_Document-updateOne) instead where possible.

**Comment by paton** [comment]
@vkarpov15 We also ran into this issue and found the deprecation documentation confusing.

We know `Model.update()` needs to be switched to updateOne or updateMany. But we had assumed that `document.update()` would still work because it's impossible (as far as I know) for that operation to affect multiple documents (since you're calling the `update()` on a single document.. unless multiple documents can have the same _id?

But it makes sense to rename it to be consistent with the deprecation of `Model.update`.. just want to point out that we also found the wording of the docs to be lacking / confusing on this topic.
